components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/internal/ClientManager.java

```diff
@@ -51,86 +67,149 @@ public class ClientManager {
 
     private static final Log LOG = LogFactory.getLog(ClientManager.class);
     private final CloseableHttpAsyncClient httpAsyncClient;
+    private final CloseableHttpClient httpClient;
+    private SSLContext sslContext;
 
     /**
      * Creates a client manager.
      *
-     * @throws WebSubAdapterException on errors while creating the http client.
+     * @throws WebSubAdapterException on errors while creating the http clients.
      */
     public ClientManager() throws WebSubAdapterException {
 
-        PoolingNHttpClientConnectionManager connectionManager;
+        // Create the http async client.
+        PoolingAsyncClientConnectionManager asyncConnectionManager;
         try {
-            connectionManager = createPoolingConnectionManager();
+            asyncConnectionManager = createAsyncPoolingConnectionManager();
         } catch (IOException e) {
             throw handleServerException(ERROR_CREATING_ASYNC_HTTP_CLIENT, e);
         }
 
-        RequestConfig config = createRequestConfig();
-        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
-        addSslContext(httpClientBuilder);
-        httpClientBuilder.setConnectionManager(connectionManager);
-        httpAsyncClient = httpClientBuilder.build();
+        LOG.info("Creating HTTP async client manager for WebSub outbound adapter.");
```

Comment:
Lets make these debug logs. Also use the name `Websubhub`

Reply:
Resolved